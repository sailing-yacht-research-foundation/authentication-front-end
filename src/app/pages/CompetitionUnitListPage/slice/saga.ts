/**
 * Root saga manages watcher lifecycle
 */

import { call, put, takeLatest } from "@redux-saga/core/effects";
import { getMany } from "services/live-data-server/event-calendars";
import { myRaceListActions } from ".";

export function* getRaces(action) {
    yield put(myRaceListActions.setIsChangingPage(true));

    const response = yield call(getMany, action.payload);

    yield put(myRaceListActions.setIsChangingPage(false));
    yield put(myRaceListActions.setPage(action.payload));

    if (response.success) {
        if (response.data?.count > 0) {
            yield put(myRaceListActions.setResults(response.data?.rows));
            yield put(myRaceListActions.setTotal(response.data?.count));
        }
    }
}

export default function* myRaceListSaga() {
    yield takeLatest(myRaceListActions.getRaces.type, getRaces);
}
