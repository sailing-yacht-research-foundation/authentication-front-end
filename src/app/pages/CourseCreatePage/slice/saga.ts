/**
 * Root saga manages watcher lifecycle
 */
 import { courseActions } from ".";
 import { call, put, takeLatest } from 'redux-saga/effects';
 import { list, create, deleteCourse, update } from "services/live-data-server/courses";
import { toast } from "react-toastify";


export function* getCourses() {
    const response = yield call(list);

    if (response.data && response.data?.rows > 0) {
        yield put(courseActions.setCourses(response.data?.rows));
        yield put(courseActions.setTotalCourses(response.data?.count));
    } else {
        toast.error('Unable to get course data');
    }
}

export function* createCourse(courseData) {
    const response = yield call(create, courseData);

    if (response.data) {
        toast.success('Course created successfully!');
    }
}

export function* deleteCourseHandler(courseId) {
    const response = yield call(deleteCourse, courseId);

    if (response.data) {
        toast.success('Successfully delete the course!');
    }
}

export function* updateCourse(data) {
    const response = yield call(update, data.courseId, data.courseData);

    if (response.data) {
        toast.success('Course updated successfully!');
    }
}

export default function* courseCreateSaga() {
    yield takeLatest(courseActions.createCourse.type, createCourse);
    yield takeLatest(courseActions.deleteCourse.type, deleteCourseHandler);
    yield takeLatest(courseActions.updateCourse.type, updateCourse);
    yield takeLatest(courseActions.getCourses.type, getCourses);
}
