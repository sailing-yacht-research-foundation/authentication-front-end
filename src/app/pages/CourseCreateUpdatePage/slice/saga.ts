/**
 * Root saga manages watcher lifecycle
 */


export default function* courseCreateSaga() {
}
