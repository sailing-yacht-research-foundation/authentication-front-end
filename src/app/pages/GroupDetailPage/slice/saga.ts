import { call, put, takeLatest } from "@redux-saga/core/effects";
import { getMembers, getAdmins, getGroupById, searchMembers } from "services/live-data-server/groups";
import { showToastMessageOnRequestError } from "utils/helpers";
import { groupDetailActions } from ".";

export function* getGroupMembers({ type, payload }) {
    const { page, groupId } = payload;
    yield put(groupDetailActions.setIsGettingMembers(true));
    const response = yield call(getMembers, groupId, page);
    yield put(groupDetailActions.setIsGettingMembers(false));

    if (response.success) {
        yield put(groupDetailActions.setMembers(response.data?.rows));
        yield put(groupDetailActions.setCurrentMemberPage(response.data?.page));
        yield put(groupDetailActions.setMemberTotal(response.data?.count));
    }
}

export function* getGroupAdmins({ type, payload }) {
    const { page, groupId } = payload;
    yield put(groupDetailActions.setIsGettingAdmins(true));
    const response = yield call(getAdmins, groupId, page);
    yield put(groupDetailActions.setIsGettingAdmins(false));

    if (response.success) {
        yield put(groupDetailActions.setAdmins(response.data?.rows));
        yield put(groupDetailActions.setCurrentAdminPage(response.data?.page));
        yield put(groupDetailActions.setAdminTotal(response.data?.count));
    }
}

export function* getGroup({ type, payload }) {
    const groupId = payload;

    yield put(groupDetailActions.setIsGettingGroup(true));
    const response = yield call(getGroupById, groupId);
    yield put(groupDetailActions.setIsGettingGroup(false));

    if (response.success) {
        yield put(groupDetailActions.setGroup(response.data));
    } else {
        yield put(groupDetailActions.setGetGroupFailed(true));
        showToastMessageOnRequestError(response.error);
    }
}

export function* searchAcceptedMembers({ type, payload }) {
    const { groupId, keyword, status } = payload;
    const response = yield call(searchMembers, groupId, keyword, status);

    if (response.success) {
        yield put(groupDetailActions.setAcceptedMemberResults(response.data.rows));
    }
}

export default function* groupSaga() {
    yield takeLatest(groupDetailActions.getAdmins.type, getGroupAdmins);
    yield takeLatest(groupDetailActions.getMembers.type, getGroupMembers);
    yield takeLatest(groupDetailActions.getGroup.type, getGroup);
    yield takeLatest(groupDetailActions.searchAcceptedMembers.type, searchAcceptedMembers);
}
