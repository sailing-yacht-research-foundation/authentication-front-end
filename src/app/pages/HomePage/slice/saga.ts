/**
 * Root saga manages watcher lifecycle
 */

import { homeActions } from ".";
import { call, put, select, takeLatest } from 'redux-saga/effects';
import { search } from "services/live-data-server/competition-units";
import { toast } from "react-toastify";
import i18next from 'i18next';
import { translations } from "locales/translations";
import { selectSearchKeyword } from "./selectors";

export function* searchRaces(params) {
    yield put(homeActions.setIsSearching(true));

    const response = yield call(search, params);
    const searchKeyword = yield select(selectSearchKeyword);

    yield put(homeActions.setIsSearching(false));

    if (response.data) {
        if (response.data?.hits?.total?.value === 0) {
            toast.info(i18next.t(translations.home_page.search_performed_no_result_found, { keyword: searchKeyword }));
        } else {
            yield put(homeActions.setResults(response.data?.hits?.hits));
            yield put(homeActions.setTotal(response.data?.hits.total?.value));
        }
    }
}

export default function* homeSaga() {
    yield takeLatest(homeActions.searchRaces.type, searchRaces);
}
