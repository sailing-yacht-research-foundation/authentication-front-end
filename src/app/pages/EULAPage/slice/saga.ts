/**
 * Root saga manages watcher lifecycle
 */

import { eulaActions } from ".";
import { call, takeLatest, select } from 'redux-saga/effects';
import { logVersion } from "services/versioning";
import { getUser } from "services/live-data-server/user";
import { selectIsAuthenticated } from "app/pages/LoginPage/slice/selectors";

export function* signEulaVersion(version) {
    const isAuthenticated = yield select(selectIsAuthenticated);

    if (!isAuthenticated) return;

    const response = yield call(getAuthorizedUser);
    
    if (response.user) yield call(logVersion, response.user?.email, 'eula', version);
}

function getAuthorizedUser() {
    return getUser();
}

export default function* loginSaga() {
    yield takeLatest(eulaActions.signEulaVersion.type, signEulaVersion);
}
