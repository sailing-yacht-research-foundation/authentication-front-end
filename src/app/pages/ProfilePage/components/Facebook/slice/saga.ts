/**
 * Root saga manages watcher lifecycle
 */

import { facebookActions } from ".";
import { call, put, takeLatest, select } from 'redux-saga/effects';
import { selectUser } from "app/pages/LoginPage/slice/selectors";
import { getFeeds } from "services/facebook";

export function* getFacebookFeeds() {
    const user = yield select(selectUser);
    const response = yield call(getFeeds, user);

    if (response.data && response.data.data)
        yield put(facebookActions.setPosts(response.data.data));
    else  yield put(facebookActions.setGetFeedsErrorState(true));
}


export function* facebookSaga() {
    yield takeLatest(facebookActions.getPosts.type, getFacebookFeeds);
}
