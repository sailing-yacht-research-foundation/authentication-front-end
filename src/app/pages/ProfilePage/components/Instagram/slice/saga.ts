/**
 * Root saga manages watcher lifecycle
 */

import { instagramActions } from ".";
import { call, put, takeLatest, select } from 'redux-saga/effects';
import { selectUser } from "app/pages/LoginPage/slice/selectors";
import { getFeeds, exchangeTokenFromCode } from "services/instagram";

export function* getInstagramFeeds() {
    const user = yield select(selectUser);
    const response = yield call(getFeeds, user);

    if (response.data && response.data.data)
        yield put(instagramActions.setPosts(response.data.data));
    else yield put(instagramActions.setGetFeedsErrorState(true));
}

export function* exchangeInstagramTokenFromCode(code) {
    const response = yield call(exchangeTokenFromCode, code);

    if (response.response) // error, change state to error 
        yield put(instagramActions.setExchangeTokenErrorState(true));

    if (response.data?.access_token) {
        yield put(instagramActions.exchangeLongLivedToken(response.data.access_token));
    }
}

export function* instagramSaga() {
    yield takeLatest(instagramActions.getPosts.type, getInstagramFeeds);
    yield takeLatest(instagramActions.exchangeTokenFromCode.type, exchangeInstagramTokenFromCode);
}
