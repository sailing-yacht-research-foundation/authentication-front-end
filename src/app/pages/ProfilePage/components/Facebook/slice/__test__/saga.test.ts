import { put, takeLatest } from 'redux-saga/effects';
import * as slice from '..';

import { facebookSaga, getFacebookFeeds } from '../saga';

describe('facebook Saga', () => {
  let feeds: any;
  let getFacebookPostsIterator: ReturnType<typeof getFacebookFeeds>;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getFacebookPostsIterator = getFacebookFeeds();
    const userDescriptor = getFacebookPostsIterator.next().value;
    expect(userDescriptor).toMatchSnapshot();

    const postsDescriptor = getFacebookPostsIterator.next().value;
    expect(postsDescriptor).toMatchSnapshot();
  });

  it('should return set posts to empty array if feed response is empty', () => {
    feeds = [];
    const putDescriptor = getFacebookPostsIterator.next(feeds).value;
    expect(putDescriptor).toEqual(
      put(slice.facebookActions.setPosts([])),
    );

    const iteration = getFacebookPostsIterator.next();
    expect(iteration.done).toBe(true);
  });
});

describe('facebook Saga Init', () => {
  const facebookIterator = facebookSaga();
  it('should start task to watch for getPosts action', () => {
    const takeLatestDescriptor = facebookIterator.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(slice.facebookActions.getPosts.type, getFacebookFeeds),
    );
  });
});
