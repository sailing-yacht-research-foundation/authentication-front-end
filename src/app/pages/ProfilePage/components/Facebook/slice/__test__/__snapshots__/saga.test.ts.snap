// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facebook Saga should exchangeLongLivedToken when response.data.access_token is defined 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`facebook Saga should exchangeLongLivedToken when response.data.access_token is defined 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`facebook Saga should exchangeLongLivedToken when response.data.access_token is defined 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`facebook Saga should return set getFeedsErrorState to true when response data is  not defined 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`facebook Saga should return set getFeedsErrorState to true when response data is  not defined 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`facebook Saga should return set getFeedsErrorState to true when response data is  not defined 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`facebook Saga should setExchangeTokenErrorState to true when response.data.access_token is not defined 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`facebook Saga should setExchangeTokenErrorState to true when response.data.access_token is not defined 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`facebook Saga should setExchangeTokenErrorState to true when response.data.access_token is not defined 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
